connector/src/it/scala/com/datastax/spark/connector/datasource/CassandraCatalogNamespaceSpec.scala

```diff
@@ -0,0 +1,104 @@
+package com.datastax.spark.connector.datasource
+
+import org.scalatest.concurrent.Eventually._
+import org.apache.spark.sql.catalyst.analysis.{NamespaceAlreadyExistsException, NoSuchNamespaceException}
+import scala.collection.JavaConverters._
+
+class CassandraCatalogNamespaceSpec extends CassandraCatalogSpecBase {
+
+  "A Cassandra Catalog Namespace Support" should "initialize successfully" in {
+    spark.sessionState.catalogManager.currentCatalog.name() should include("Catalog cassandra")
+  }
+
+  it should "list available keyspaces" in {
+    val currentKeyspaces = spark.sql("SHOW DATABASES").collect().map(_.getString(0))
+    val existingKeyspaces = getMetadata().getKeyspaces().keySet().asScala.map(_.asInternal())
+    currentKeyspaces should contain allElementsOf(existingKeyspaces)
+
+  }
+
+  it should "list a keyspaces in a keyspace"  in {
+    spark.sql(s"CREATE DATABASE IF NOT EXISTS $defaultKs WITH DBPROPERTIES (class='SimpleStrategy',replication_factor='5')")
+    waitForKeyspaceToExist(defaultKs, true)
+    //Using Catalog and DefaultKS triggers listNamespaces(namespace) pathway
+    val result = spark.sql(s"SHOW NAMESPACES FROM $catalogName.$defaultKs").collect
+    result shouldBe empty
+  }
+
+  it should "give no keyspace found when looking up a keyspace which doesn't exist" in {
+    intercept[NoSuchNamespaceException]{
+      spark.sql(s"SHOW NAMESPACES FROM $catalogName.fakenotreal").collect
+    }
+  }
+
+  it should "be able to create a new keyspace" in {
+    dropKeyspace(defaultKs)
+    waitForKeyspaceToExist(defaultKs, false)
+    spark.sql(s"CREATE DATABASE $defaultKs WITH DBPROPERTIES (class='SimpleStrategy',replication_factor='5')")
+    waitForKeyspaceToExist(defaultKs, true)
+    spark.sql(s"DESCRIBE DATABASE EXTENDED $defaultKs").show
+  }
+
+  it should "give a sensible error when trying to create a keyspace without a replication class" in {
+    dropKeyspace(defaultKs)
+    waitForKeyspaceToExist(defaultKs, false)
+    val exception = intercept[CassandraCatalogException]{
+      spark.sql(s"CREATE DATABASE $defaultKs WITH DBPROPERTIES (replication_factor='5')")
+    }
+    exception.getMessage should include("requires a class")
+  }
+
+  it should "give a sensible error when trying to create a keyspace with an invalid replication strategy" in {
+    dropKeyspace(defaultKs)
+    waitForKeyspaceToExist(defaultKs, false)
+    val exception = intercept[CassandraCatalogException]{
+      spark.sql(s"CREATE DATABASE $defaultKs WITH DBPROPERTIES (class='foobar',replication_factor='5')")
+    }
+    exception.getMessage should include("Unknown keyspace replication strategy")
+  }
+
+  it should "throw a keyspace exists exception when creating a keyspace that already exists" in {
+    spark.sql(s"CREATE DATABASE IF NOT EXISTS $defaultKs WITH DBPROPERTIES (class='SimpleStrategy',replication_factor='5')")
+    intercept[NamespaceAlreadyExistsException]{
```

Comment:
Need a waitForKeyspaceToExist() here to confirm that we should get the exception we expect below?

Reply:
The intercept will fail if no exception is thrown, so I think that covers it